Zotero/Scenes/Detail/ItemDetail/Views/ItemDetailViewController.swift

```diff
@@ -53,36 +69,77 @@ final class ItemDetailViewController: UIViewController {
     override func viewDidLoad() {
         super.viewDidLoad()
 
-        self.navigationController?.setToolbarHidden(true, animated: false)
-        self.setupCollectionViewHandler()
-        self.setupFileObservers()
-
-        self.viewModel.stateObservable
-                      .observe(on: MainScheduler.instance)
-                      .subscribe(onNext: { [weak self] state in
-                          self?.update(to: state)
-                      })
-                      .disposed(by: self.disposeBag)
+        navigationController?.setToolbarHidden(true, animated: false)
+        setupFileObservers()
 
-        self.viewModel.process(action: .loadInitialData)
+        viewModel.stateObservable
+            .observe(on: MainScheduler.instance)
+            .subscribe(onNext: { [weak self] state in
+                self?.update(to: state)
+            })
+            .disposed(by: disposeBag)
+
+        func setupFileObservers() {
+            NotificationCenter.default.rx
+                .notification(.attachmentFileDeleted)
+                .observe(on: MainScheduler.instance)
+                .subscribe(onNext: { [weak viewModel] notification in
+                    guard let viewModel, let notification = notification.object as? AttachmentFileDeletedNotification else { return }
+                    viewModel.process(action: .updateAttachments(notification))
+                })
+                .disposed(by: disposeBag)
+
+            NotificationCenter.default.rx
+                .notification(UIApplication.willEnterForegroundNotification)
+                .observe(on: MainScheduler.instance)
+                .subscribe(onNext: { [weak self] _ in
+                    guard let self else { return }
+                    // Need to reload data to current state before going back to foreground. iOS reloads the collection view when going to foreground with current snapshot. When
+                    // editing text fields we don't update the snapshot (so that the cell is not reloaded while typing), so the edited fields are reset to previous state.
+                    collectionViewHandler.reloadAll(to: viewModel.state, animated: false)
+                })
+                .disposed(by: disposeBag)
+
+            controllers.userControllers?.fileDownloader.observable
+                .observe(on: MainScheduler.instance)
+                .subscribe(onNext: { [weak self] update in
+                    guard let self else { return }
+                    viewModel.process(action: .updateDownload(update))
+
+                    guard viewModel.state.attachmentToOpen == update.key else { return }
+
+                    // 🍎 check logic of processes
```

Comment:
Any reason to move from `// TODO` to `// 🍎`? :) Are these which you want to solve before closing this PR?

Reply:
I somehow didn't delete all of my temporary comments & prints (I use an emoji for easier search). Sorry about that, fixed in https://github.com/zotero/zotero-ios/pull/1021/commits/a0358fce184823bf9b608541d5d6e69838c627bc